feat: add smart contract deployment docs

nakul1010 opened the pull request:
The Pr adds the following things
- Adds Smart contract deployment guide under `build`
  - Foundry
  - Remix

nakul1010 commented:
Any reason to include docs folder under gitignore fil? Currently removed it but can add back if necessary and only allow deploy folder

nakul1010 commented:
Any reason to include docs folder under gitignore fil? Currently removed it but can add back if necessary and only allow deploy folder

nakul1010 commented:
Any reason to include docs folder under gitignore? Currently removed it but can only also allow deploy folder and ignore other docs filder content.

nud3l reviewed (changes requested):
Looks like this PR adds a bunch of the SDK build results to git, which should be ignored. Can you please remove these?

nud3l commented on docs/docs/build/deploy/foundry.md:
```diff
@@ -0,0 +1,152 @@
+
+- Checkout [testnet explorer](https://explorerl2-fluffy-bob-7mjgi9pmtg.t.conduit.xyz) to get contract details using Transaction hash. 
+
+![Contract details on Explorer Image](../../../static/img/deployed_contract_on_foundry.png)
```
Instead of putting images in the static folder, they should be directly in the same folder as the docs. See, e.g., https://github.com/bob-collective/bob/tree/master/docs/docs/learn/introduction

nud3l commented:
> Any reason to include docs folder under gitignore? Currently removed it but can add back if necessary and only allow deploy folder

No good reason, good call to remove that part from the gitignore file.

nakul1010 commented:
> Looks like this PR adds a bunch of the SDK build results to git, which should be ignored. Can you please remove these?

Removed the generated `docs/docs/api` also added it in `gitignore`
